modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/MavenDeployableProjectModelBuilder.java

```diff
@@ -159,36 +170,47 @@ private Supplier<MuleDeployableModel> getDeployableModelResolver(ArtifactCoordin
 
   private boolean shouldEditApplicationModel(MuleApplicationModel applicationModel) {
     return (exportAllResourcesAndPackagesIfEmptyLoaderDescriptor
-        && applicationModel.getClassLoaderModelLoaderDescriptor() == null) || includeTestDependencies;
+        && applicationModel.getClassLoaderModelLoaderDescriptor() == null) || includeTestDependencies
+        || applicationModel.getConfigs() == null;
```

Comment:
In case the field is empty it means that the `configs` field was present in the `mule-artifact.json` (although empty) and I think we should honour its value in that case.